dix/req.c

```diff
@@ -1826,6 +1827,15 @@ static int dix_cas_rops_send(struct m0_dix_req *req)
 		rc = m0_dtx0_close(req->dr_dtx);
 		if (rc != 0)
 			return M0_ERR(rc);
+		/*
+		 * It is safe to set EXECUTED dtx state for those
+		 * participants that experience transient failure,
+		 * it allows to trigger EXECUTED-ALL logic.
+		 */
+		for (pa_idx = cas_rop_tlist_length(&rop->dg_cas_reqs);
```

Comment:
Here, can we use:

pa_idx = rop->dg_cas_reqs_nr;

?

Reply:
I need to understand:
why are the CAS_REQ not created for pa_idx in [cas_rop_tlist_length(&rop->dg_cas_reqs), req->dr_dtx->tx_dtx->dd_txd.dtd_ps.dtp_nr)?

because we already know they are OFFLINE?

Reply:
1. No, we need to use cas_rop_tlist_length(&rop->dg_cas_reqs) instead of rop->dg_cas_reqs_nr as rop->dg_cas_reqs_nr may be less then list length due to sending errors, we need to get the position of the first OFFLINE participant here.

2. Yes, we don't send CAS requests to OFFLINE participants, that is checked using dix_pg_unit_skip().

Reply:
OK